File: src/com/suushiemaniac/cubing/bld/model/enumeration/PieceType.java
package com.suushiemaniac.cubing.bld.model.enumeration;

import com.suushiemaniac.cubing.alglib.lang.NotationReader;

public interface PieceType {
    int getTargetsPerPiece();

    int getNumPieces();

    int getNumPiecesNoBuffer();

    int getNumAlgs();

    String name();

    String humanName();

	String mnemonic();

    NotationReader getReader();
}

File: src/main/kotlin/com/suushiemaniac/cubing/bld/model/PieceType.kt
package com.suushiemaniac.cubing.bld.model

data class PieceType(val name: String, val permutations: Int, val orientations: Int) {
    val permutationsNoBuffer: Int
        get() = this.permutations - 1

    val numTargets: Int
        get() = this.permutations * this.orientations

    val humanName: String
        get() = this.name.split("_").joinToString("") { it.toLowerCase().capitalize() }
}
